From c273bc473cafefd385db070fe82f962a32187234 Mon Sep 17 00:00:00 2001
From: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Date: Thu, 13 Oct 2016 17:01:56 -0400
Subject: [PATCH 1307/1722] drm/amd/dal: MST monitor hotplug hang.

Don't try to read EDID when there is no physical sink on a downstream
port.

Change-Id: I656877e77efeeb1150247e1114d161c13251e736
Signed-off-by: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
index bc3f72e..b710df8 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
@@ -338,7 +338,8 @@ static void dm_dp_mst_hotplug(struct drm_dp_mst_topology_mgr *mgr)
 	drm_modeset_lock_all(dev);
 	list_for_each_entry(connector, &dev->mode_config.connector_list, head) {
 		aconnector = to_amdgpu_connector(connector);
-		if (aconnector->port && !aconnector->dc_sink) {
+		if (aconnector->port && aconnector->port->pdt != DP_PEER_DEVICE_NONE
+				&& !aconnector->dc_sink) {
 			/*
 			 * This is plug in case, where port has been created but
 			 * sink hasn't been created yet
-- 
2.7.4

